unit/solvers/smt2_incremental/object_tracking.cpp

```diff
@@ -27,11 +27,6 @@ TEST_CASE("find_object_base_expression", "[core][smt2_incremental]")
     rowt{
       "Address of struct member",
       {member_exprt{object_base, "baz", unsignedbv_typet{8}}, pointer_type}},
-    rowt{
-      "Address of index of struct member",
-      {index_exprt{
-         member_exprt{object_base, "baz", unsignedbv_typet{8}}, index},
-       pointer_type}},
```

Comment:
@NlightNFotis @thomasspriggs Could you please review this code (and the surrounding lines) to clarify what is meant to be going on here? Multiple type-inconsistent expressions appear to be formed here.

Reply:
`GENERATE_REF` is being used to generate a set of tests where the `find_object_base_expression` function is called with various expressions and is expected to return `object_base`. The `find_object_base_expression` function is analogous to part of the implementation of `pointer_logict::add_object`. The unit test currently works because the `find_object_base_expression` function does not currently depend on the `.type()` of the expressions or their sub-expressions and because the constructors of `index_exprt` and `member_exprt` do not check the type of `object_base`.

If you change the currently deleted lines of code to the following, then it should make the `index_exprt` sufficiently type-consistent to work with this PR -
```
    rowt{
      "Address of index of struct member",
      {index_exprt{
         member_exprt{object_base, "baz", base_type}, index},
       pointer_type}},
```

I acknowledge that this does not make the instances of `member_exprt` type-consistent. They could be made type-consistent as well but this would require expanding the test with differently typed base_object expressions for each generated test, using various `struct_typet`s where appropriate.